generator/C/include_v2.0/mavlink_helpers.h

```diff
@@ -742,18 +742,23 @@ MAVLINK_HELPER uint8_t mavlink_frame_char_buffer(mavlink_message_t* rxmsg,
 
 	case MAVLINK_PARSE_STATE_GOT_PAYLOAD: {
 		const mavlink_msg_entry_t *e = mavlink_get_msg_entry(rxmsg->msgid);
-		uint8_t crc_extra = e?e->crc_extra:0;
-		mavlink_update_checksum(rxmsg, crc_extra);
-		if (c != (rxmsg->checksum & 0xFF)) {
+		if (e == NULL) {
+			// Message not found in CRC_EXTRA table.
 			status->parse_state = MAVLINK_PARSE_STATE_GOT_BAD_CRC1;
```

Comment:
you also need to set rxmsg->ck[0] to the byte from the packet (c) so that this code works correctly for forwarding the message: ```
	if (status->msg_received == MAVLINK_FRAMING_BAD_CRC) {
		/*
		  the CRC came out wrong. We now need to overwrite the
		  msg CRC with the one on the wire so that if the
		  caller decides to forward the message anyway that
		  mavlink_msg_to_send_buffer() won't overwrite the
		  checksum
		 */
            if (r_message != NULL) {
                r_message->checksum = rxmsg->ck[0] | (rxmsg->ck[1]<<8);
            }
	}
```

Reply:
I see, so you can still forward even if you can't fully parse a message. Do you have an example of such a usage?

Reply:
https://github.com/ArduPilot/ardupilot/blob/7d426f4741c7f77b03c26b98899e5d3d89503e8f/libraries/GCS_MAVLink/GCS_Common.cpp#L1889 is one related implementation

Reply:
Got it, thanks. That's interesting.

I will update as suggested to fix that case.

Reply:
Fix pushed.